modules/ml/src/main/java/org/apache/ignite/ml/tree/impurity/util/StepFunction.java

```diff
@@ -47,6 +49,19 @@ public StepFunction(double[] x, T[] y) {
         sort(x, y, 0, x.length - 1);
     }
 
+    @SuppressWarnings("unchecked")
+    public StepFunction(ArrayList<Double> x, ArrayList<T> y, Class<T> yClass) {
+        assert x.size() == y.size() : "Argument and value arrays have to be the same length";
+
+        this.x = new double[x.size()];
+        this.y = (T[])Array.newInstance(yClass, y.size());
```

Comment:
Does it make sense to update constructor signature so that it accepts arrays and make this list to array transformation before the construction? Will look better from my perspective.

Reply:
I delete this, because of  "We know size of these arrays, don't we?"